ipn/ipnlocal/serve.go

```diff
@@ -575,6 +575,20 @@ type reverseProxy struct {
 	h2cTransport lazy.SyncValue[http.RoundTripper]
 }
 
+// closeIdleConnections closes any connections open for either of its transports
+func (rp *reverseProxy) closeIdleConnections() {
+	if h2cT := rp.h2cTransport.Get(func() http.RoundTripper {
```

Comment:
check at the other call sites if it is `nil` and return an error like "connection refused: shutting down"

Reply:
maybe put an `closed atomic.Bool` on the proxy.

Reply:
> maybe put an closed atomic.Bool on the proxy.

nice